### ISSUE TITLE
No more outlines since V4

### ISSUE THREAD

**Issue by whitespacecode** [Tresjs/tres#742]
### Describe the bug

Using the [@pmndrs/vanilla](https://github.com/pmndrs/drei-vanilla) package to create an Outline around the Mesh fails in V4

On V3.9
![image](https://github.com/Tresjs/tres/assets/37139936/61ed3675-ae56-4692-83b7-e15763cca2cc)
On V4.x
![image](https://github.com/Tresjs/tres/assets/37139936/f9834124-e0cb-43ef-b348-35650d0794e2)

No console error's..
I used @hawk86104 [example](https://github.com/hawk86104/icegl-three-vue-tres/blob/master/src/plugins/basic/components/outlineCom.vue) to create this and noticed the outline's break in V4

### Reproduction

https://stackblitz.com/edit/tresjs-basic-9b2xts?file=src%2Fcomponents%2FTheExperience.vue

### Steps to reproduce

Change the package.json so Tresjs uses V4 and the outline will be gone

### System Info

```shell
/
```

### Used Package Manager

npm

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/Tresjs/tres/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/Tresjs/tres/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://tresjs.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/tresjs/tres/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by alvarosabu** [comment]
Hi there @whitespacecode I'm really not familiar on that package so maybe @hawk86104 can support us here

**Comment by alvarosabu** [comment]
Or why not using Tresjs Postprocessing package directly https://post-processing.tresjs.org/guide/effects/outline.html ?

**Comment by whitespacecode** [comment]
Wait.. @alvarosabu There is a post-processing.tresjs!?
I will look into that tomorrow. Thanks

**Comment by hawk86104** [comment]
@alvarosabu @whitespacecode 
It's been a while since we last got in touch.🤓
Sorry, I have been busy recently with the ecosystem of Tress in China, mainly improving small functionalities across various industries: http://opensource.icegl.cn/

Regarding the version dependency of Tress, I have not upgraded to 4.x because the upgrade would cause many issues with the previous functionalities. I did a brief check and found quite a few problems, so I plan to wait until the 4.x version is more complete and stable before upgrading.
As for the outline effect, we are using a shader-based method rather than post-processing.

**Comment by whitespacecode** [comment]
I noticed when i use the outlines package directly on the mesh everything works.
It should normally extract the geometry from its parent and that's where it's failing suddenly.  So i'm not sure what changed in V4 to the `TresMesh`

```ts
//This works fine on V4
const outlines = Outlines()
const mesh = new THREE.Mesh(geometry, material)
mesh.add(outlines.group)
```

**Comment by hawk86104** [comment]
@whitespacecode  @alvarosabu 
https://github.com/Tresjs/tres/issues/766#issuecomment-2222003076
The issue arises from the use of a proxy for ‘primitive’ in ‘setPrimitiveObject’, leading to copies and shallow copies that result in the loss of the source object’s properties and methods. Awaiting an official fix.

and this is tvt v4.x branch：
https://github.com/hawk86104/icegl-three-vue-tres/tree/V4.x

**Comment by whitespacecode** [comment]
> @whitespacecode @alvarosabu [#766 (comment)](https://github.com/Tresjs/tres/issues/766#issuecomment-2222003076) The issue arises from the use of a proxy for ‘primitive’ in ‘setPrimitiveObject’, leading to copies and shallow copies that result in the loss of the source object’s properties and methods. Awaiting an official fix.
> 
> and this is tvt v4.x branch： https://github.com/hawk86104/icegl-three-vue-tres/tree/V4.x

Thanks for your fix and insight! I prefer the shader-based outline more. 
Alvarosabu did some changes to primitive recently https://github.com/Tresjs/tres/pull/764

**Comment by whitespacecode** [comment]
This is fixed in V4.2
